TrafficCapture/captureKafkaOffloader/src/main/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/KafkaCaptureFactory.java

```diff
@@ -104,62 +96,47 @@ public CodedOutputStreamWrapper createStream() {
-                        topicNameForTraffic, recordId, kafkaRecord.value().length);
-
-                // Async request to Kafka cluster
-                producer.send(kafkaRecord, handleProducerRecordSent(cf, recordId, flushContext));
-                metricsLogger.atSuccess(MetricsEvent.RECORD_SENT_TO_KAFKA)
-                        .setAttribute(MetricsAttributeKey.CHANNEL_ID, connectionId)
-                        .setAttribute(MetricsAttributeKey.TOPIC_NAME, topicNameForTraffic)
-                        .setAttribute(MetricsAttributeKey.SIZE_IN_BYTES, kafkaRecord.value().length)
-                        .setAttribute(MetricsAttributeKey.REQUEST_ID, recordId).emit();
-                return cf;
-            } catch (Exception e) {
-                metricsLogger.atError(MetricsEvent.RECORD_FAILED_TO_KAFKA, e)
-                        .setAttribute(MetricsAttributeKey.CHANNEL_ID, connectionId)
-                        .setAttribute(MetricsAttributeKey.TOPIC_NAME, topicNameForTraffic).emit();
-                throw e;
-            }
+
+            String recordId = String.format("%s.%d", connectionId, index);
+            var byteBuffer = osh.byteBuffer;
+            ProducerRecord<String, byte[]> kafkaRecord = new ProducerRecord<>(topicNameForTraffic, recordId,
+                    Arrays.copyOfRange(byteBuffer.array(), 0, byteBuffer.position()));
+            // Used to essentially wrap Future returned by Producer to CompletableFuture
+            var cf = new CompletableFuture<RecordMetadata>();
+            log.debug("Sending Kafka producer record: {} for topic: {}", recordId, topicNameForTraffic);
+
+            var flushContext = rootScope.createKafkaRecordContext(telemetryContext,
+                    topicNameForTraffic, recordId, kafkaRecord.value().length);
+
+            // Producer Send will block on calls such as retrieving cluster metadata, running fully async
+            return CompletableFuture.supplyAsync(() -> {
+                    try {
+                        /*
+                         * The default KafkaProducer comes with built-in retry and error-handling logic that suits many cases. From the
+                         * documentation here for retry: https://kafka.apache.org/35/javadoc/org/apache/kafka/clients/producer/KafkaProducer.html
+                         * "If the request fails, the producer can automatically retry. The retries setting defaults to Integer.MAX_VALUE,
+                         * and it's recommended to use delivery.timeout.ms to control retry behavior, instead of retries."
+                         *
+                         * Apart from this the KafkaProducer has logic for deciding whether an error is transient and should be
+                         * retried or not retried at all: https://kafka.apache.org/35/javadoc/org/apache/kafka/common/errors/RetriableException.html
+                         * as well as basic retry backoff
```

Comment:
I provide some alternate code below but didn't include this comment.  Please make sure to continue to include this.